Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/DocumentPropertyEditor/PrefabAdapter.cpp

```diff
@@ -38,17 +46,36 @@ namespace AzToolsFramework::Prefab
         AZ::Interface<PrefabAdapterInterface>::Unregister(this);
     }
 
-    void PrefabAdapter::AddPropertyHandlerIfOverridden(
-        AZ::DocumentPropertyEditor::AdapterBuilder* adapterBuilder, const AZ::Dom::Path& relativePathFromEntity, AZ::EntityId entityId) 
+    void PrefabAdapter::AddPropertyIconAndLabel(
+        AZ::DocumentPropertyEditor::AdapterBuilder* adapterBuilder,
+        const AZStd::string_view labelText,
+        const AZ::Dom::Path& relativePathFromEntity,
+        AZ::EntityId entityId)
     {
-        if (m_prefabOverridePublicInterface->AreOverridesPresent(entityId, relativePathFromEntity.ToString()))
+        using PrefabPropertyEditorNodes::PrefabOverrideLabel;
+
+        // Do not show override visualization on container entities.
+        if (m_prefabPublicInterface->IsInstanceContainerEntity(entityId))
         {
-            adapterBuilder->BeginPropertyEditor<PrefabPropertyEditorNodes::PrefabOverrideProperty>();
-            adapterBuilder->Attribute(AZ::DocumentPropertyEditor::Nodes::PropertyEditor::SharePriorColumn, true);
-            adapterBuilder->Attribute(AZ::DocumentPropertyEditor::Nodes::PropertyEditor::UseMinimumWidth, true);
-            adapterBuilder->Attribute(
-                AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Alignment,
-                AZ::DocumentPropertyEditor::Nodes::PropertyEditor::Align::AlignLeft);
+            adapterBuilder->BeginPropertyEditor<PrefabOverrideLabel>();
+            adapterBuilder->Attribute(PrefabOverrideLabel::Text, labelText);
+            adapterBuilder->Attribute(PrefabOverrideLabel::IsOverridden, false);
+            adapterBuilder->EndPropertyEditor();
```

Comment:
code duplication can be avoided if you combine this if check with the check in line 70.